### PR TITLE
fix(table): used the default values from the sub components instead o…

### PR DESCRIPTION
**Pull request by peterpham-kyndryl**
…f the table. Only apply new values if they are set. Fix multi selection when pre-checked values are passed in

## Summary

Brief summary describing the changes here.

## ADO Story or GitHub Issue Link

Link here (if applicable).

## Notes

- Detailed change notes
- can go here

## To Do

- Anything else
- left to do

## Checklist

- [ ] Used Conventional Commit messages as outlined in the contributing guide.
  - [ ] Noted breaking changes (if any).
- [ ] Documented/updated all props, events, slots, parts, etc with JSDoc.
  - [ ] Ran the `analyze` command to update Storybook docs.
- [ ] Added/updated Stories with controls to cover all variants.
- [ ] Ran `test` locally to address any failures.
- [ ] Added/updated the Figma link for the Story's Design tab.
- [ ] Added any new component exports to the src/index.ts file

## Testing Instructions

Provide guidance to reviewers/testers here.

## Screenshots

(if any)

### CODE REVIEW

**Review by maulik-thaker1 — CHANGES_REQUESTED** [review]
Here its shows **boolean** & **undefined** it should be **boolean** IMO

Screenshot 2024-04-23 at 5 57 49 PM

**Comment by maulik-thaker1** [comment]
Here as well. Need to do `npm run analyze` once

Screenshot 2024-04-23 at 6 01 42 PM

**Comment by peterpham-kyndryl** [comment]
> Here its shows **boolean** & **undefined** it should be **boolean** IMO
> 
> Screenshot 2024-04-23 at 5 57 49 PM

Yes, they're boolean, but they are optional. Looks like Storybook auto docs is not generating the optional props properly. I wanted these props to be interpreted as that these are optional, but if they're passed in they have to be boolean. . 
Anyways, I agree that this has to be fixed.

File: src/components/reusable/table/table-header.ts [hunk 1]
```diff
@@ -157,8 +179,21 @@ export class TableHeader extends LitElement {
...
 
-  override updated() {
+  override updated(changedProperties: PropertyValues) {
     this.getTextContent();
+
+    super.updated(changedProperties);
+    if (this.maxWidth && changedProperties.has('maxWidth')) {
+      this.style.setProperty('--kyn-th-max-width', this.maxWidth);
+    }
+
```
Comment (maulik-thaker1) [inline src/components/reusable/table/table-header.ts:189]: IMO no need to check `this.maxWidth` here as it's already covered in `changedProperties.has('maxWidth')`. So I guess your if condition will be 

`if(changedProperties.has('maxWidth'){ ... }`

Same applies for other if conditions also. [`this.minWidth` and `this.width`]

**Review by maulik-thaker1 — CHANGES_REQUESTED** [review]

File: src/components/reusable/table/table.ts [hunk 2]
```diff
@@ -25,35 +25,35 @@ export class Table extends LitElement {
    * selectable using checkboxes.
    */
   @property({ type: Boolean })
-  checkboxSelection = false;
+  checkboxSelection?: boolean;
```
[the remark this answers is not in the document]
Comment (peterpham-kyndryl) [reply src/components/reusable/table/table.ts:29]: Like I mentioned previously, these are optional so the default values will be undefined. When they're not set, the sub-components will use their own default values. Those values will only be overridden if these values are really set by the users.
The way updated function works is that even if the user doesn't set a new value, it's still invoked at the first time when the values are set from undefined to the default values. And this is not what I want to happen, so I need them to be undefined by default, not false.

File: src/components/reusable/table/table-header.ts (hunk 1 above)
Comment (peterpham-kyndryl) [reply src/components/reusable/table/table-header.ts:189]: changedProperties.has('maxWidth') only checks if there is a change of maxWidth, meanwhile this.maxWidth is checking for a truthy (i.e., it has a value other than undefined, null, 0, NaN, an empty string, or false). If this.width is falsy, the property is not set, which can prevent potential issues or errors.

**Comment by peterpham-kyndryl** [comment]
Closing since changes are included in this PR https://github.com/kyndryl-design-system/shidoka-applications/pull/157

**Pull request closed without merging**